lightning/src/ln/channelmanager.rs

```diff
@@ -5173,49 +5197,60 @@ where
 		let mut peer_state_lock = peer_state_mutex.lock().unwrap();
 		let peer_state = &mut *peer_state_lock;
 		let is_only_peer_channel = peer_state.total_channel_count() == 1;
-		match peer_state.inbound_v1_channel_by_id.entry(temporary_channel_id.clone()) {
-			hash_map::Entry::Occupied(mut channel) => {
-				if !channel.get().is_awaiting_accept() {
-					return Err(APIError::APIMisuseError { err: "The channel isn't currently awaiting to be accepted.".to_owned() });
+
+		// Find (and remove) the channel in the unaccepted table. If it's not there, something weird is
+		// happening and return an error. N.B. that we create channel with an outbound SCID of zero so
+		// that we can delay allocating the SCID until after we're sure that the checks below will
+		// succeed.
+		let mut channel = match peer_state.inbound_channel_request_by_id.remove(temporary_channel_id) {
+			Some(unaccepted_channel) => {
+				let best_block_height = self.best_block.read().unwrap().height();
+				InboundV1Channel::new(&self.fee_estimator, &self.entropy_source, &self.signer_provider,
+					counterparty_node_id.clone(), &self.channel_type_features(), &peer_state.latest_features,
+          &unaccepted_channel.open_channel_msg, user_channel_id, &self.default_configuration, best_block_height,
```

Comment:
Now that `user_channel_id` is actually picked by the user for manually accepted channels, it might be good to change some of the docs around it that say it's randomized for inbound channels. I found this mainly in `ChannelDetails`, and a couple variants in `Event`